components/script/dom/macros.rs

```diff
@@ -632,3 +632,33 @@ macro_rules! handle_potential_webgl_error {
         handle_potential_webgl_error!($context, $call, ());
     };
 }
+
+macro_rules! impl_rare_data (
+    ($type:ty) => (
+        fn init_rare_data(&self) {
+            let mut rare_data = self.rare_data.borrow_mut();
+            if rare_data.is_none() {
+                *rare_data = Some(Default::default());
+            }
+        }
+
+        fn rare_data(&self) -> Ref<Option<Box<$type>>> {
+            self.init_rare_data();
```

Comment:
This doesn't make sense. The point of the rare data is that unless you need to initialize something in there you should not allocate it. So no need to call `init_rare_data` here. Callers of course need to deal with it...